airflow/executors/celery_executor.py

```diff
@@ -264,30 +284,45 @@ def sync(self):
                 )
                 continue
             key, state = key_and_state
-            try:
-                if self.last_state[key] != state:
-                    if state == celery_states.SUCCESS:
-                        self.success(key)
-                        del self.tasks[key]
-                        del self.last_state[key]
-                    elif state == celery_states.FAILURE:
-                        self.fail(key)
-                        del self.tasks[key]
-                        del self.last_state[key]
-                    elif state == celery_states.REVOKED:
-                        self.fail(key)
-                        del self.tasks[key]
-                        del self.last_state[key]
-                    else:
-                        self.log.info("Unexpected state: %s", state)
-                        self.last_state[key] = state
-            except Exception:
-                self.log.exception("Error syncing the Celery executor, ignoring it.")
-
-    def end(self, synchronous=False):
+            self.update_task_state(key, state)
+
+    # noinspection PyUnreachableCode
```

Comment:
What does it think is unreachable in here? 🤔

Reply:
I removed it. The problem was that PyCharm thinks that "fail()" a few lines below is something that will throw an exception. ¯\_(ツ)_/¯ . Just by the method name. I do not want to change the BaseExecutor API because of that. But I removed the comments as it is indeed misleading.